src/firefly/java/edu/caltech/ipac/firefly/server/query/AsyncTapQuery.java

```diff
@@ -29,19 +31,38 @@ public class AsyncTapQuery extends UwsJobProcessor {
     public static final String SVC_URL = "serviceUrl";
     public static final String UPLOAD_TNAME = "adqlUploadSelectTable";
 
-
+    // links taken from src/firefly/js/ui/tap/TapKnownServices.js#makeServices
+    private static final List<String> SVC_RUNID_NOT_SUPPORTED = Arrays.asList(
+            "https://irsa.ipac.caltech.edu/TAP",                // Return exception, BAD_REQUEST: RUNID not implemented
+            "https://exoplanetarchive.ipac.caltech.edu/TAP/",   // Bad implementation. it replaces it with its own identifier, e.g. 109294
+            "https://koa.ipac.caltech.edu/TAP/",                // Bad implementation. it replaces it with its own identifier, e.g. 109294
+            "https://heasarc.gsfc.nasa.gov/xamin/vo/tap",       // Accepted the parameter, but did not return its value
+            "https://vao.stsci.edu/CAOMTAP/TapService.aspx",    // Accepted the parameter, but did not return its value
+            "https://gea.esac.esa.int/tap-server/tap",          // Accepted the parameter, but did not return its value
+            "https://dc.g-vo.org/tap",                          // Accepted the parameter, but did not return its value
+            "https://archives.esac.esa.int/hsa/whsa-tap-server/tap"     // Accepted the parameter, but did not return its value
+    );
```

Comment:
Maybe only exceptions should only be on this list and not ones that ignore the parameter. If they fix it then we will still not send the runid.

Reply:
Sure.  I can limit the list to only IRSA.  I'll keep the rest as comments so we know what to expect from these services.